plugins/sigma/src/parse.cpp

```diff
@@ -321,18 +321,47 @@ caf::expected<expression> parse_search_id(const data& yaml) {
         } else if (*i == "utf16") {
           return caf::make_error(ec::unimplemented, //
                                  "utf16 not yet implemented");
+        } else if (*i == "startswith") {
+          auto to_re = [](const data& d) -> caf::expected<data> {
+            auto f = detail::overload{[](const auto& x) -> caf::expected<data> {
+              auto str = vast::detail::control_char_escape(to_string(x));
+              str = fmt::format("^{}.*", str);
+              auto result = pattern::make(str, {.case_insensitive = true});
+              if (!result)
+                return std::move(result.error());
+              return std::move(*result);
+            }};
+            return caf::visit(f, d);
+          };
+          transforms.emplace_back(to_re);
+        } else if (*i == "endswith") {
+          op = relational_operator::equal;
+          auto to_re = [](const data& d) -> caf::expected<data> {
+            auto f = detail::overload{[](const auto& x) -> caf::expected<data> {
+              auto str = vast::detail::control_char_escape(to_string(x));
+              str = fmt::format(".*{}$", str);
```

Comment:
Logically, wouldn't it make sense to wrap the translated pattern into `.*{}$` _after_ you've converted it into a pattern? Otherwise you will treat it as string input, no?